### PR TITLE
Verification of two-phase flow in porous media with Buckley-Leverett problem

### PR DESCRIPTION
**Pull request by jhuang2601**
This PR 

- [x] documents the analytical verification of CO2 core flood experiment with Buckley-Leverett theory;

image

- [x] adds `buckleyLeverett_smoke` to the integratedTests and checks the running on Pecan.
       [GEOSX/integratedTests#248](https://github.com/GEOSX/integratedTests/pull/248)

### CODE REVIEW

**Review by herve-gross — APPROVED** [review]
Great example, useful for training. Thank you!

File: src/docs/sphinx/advancedExamples/validationStudies/carbonStorage/buckleyLeverett/Example.rst [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+#################################################################################
+ Verification of CO2 Core Flood Experiment with Buckley-Leverett Solution
+#################################################################################
+
+
+**Context**
+
+In this example, we simulate a CO2 core flood experiment representing immiscible transport of two-phase
+flow (CO2 and water) through porous media `(Ekechukwu et al., 2022) <https://arxiv.org/pdf/2203.05021.pdf>`__. This problem is solved using the multiphase flow solver in GEOSX to obtain the temporal evolution of saturation along the flow direction, and verified against the Buckley-Leverett analytical solutions `(Arabzai and Honma, 2013)  <http://bulletin.soe.u-tokai.ac.jp/english_vol38/20_25.pdf>`__.
```
Comment (francoishamon) [inline src/docs/sphinx/advancedExamples/validationStudies/carbonStorage/buckleyLeverett/Example.rst:12]: Maybe also cite the original paper:
S.E. Buckley and M.C. Leverett (1942). ["Mechanism of fluid displacements in sands"](https://www.onepetro.org/journal-paper/SPE-942107-G). Transactions of the AIME (146): 107–116.

**Review by francoishamon — APPROVED** [review]
Looks great, thanks!

File: src/docs/sphinx/advancedExamples/validationStudies/carbonStorage/buckleyLeverett/Example.rst (hunk 1 above)
Comment (jhuang2601) [reply src/docs/sphinx/advancedExamples/validationStudies/carbonStorage/buckleyLeverett/Example.rst]: Sure, add it

**Comment by TotoGaz** [comment]
@jhuang2601 @francoishamon So this is not added into the integratedTests, we're OK with this?
As a result we do not need to rebaseline.

**Comment by jhuang2601** [comment]
> @jhuang2601 @francoishamon So this is not added into the integratedTests, we're OK with this? As a result we do not need to rebaseline.

@TotoGaz We need to rebaseline as new smoke test is added into integratedTest. But I cannot run it on Quartz and have requested @francoishamon to do it for me.

**Comment by TotoGaz** [comment]
> @TotoGaz We need to rebaseline as new smoke test is added into integratedTest. But I cannot run it on Quartz and have requested @francoishamon to do it for me.

Yes, I was trying to give it a go myself to purge the merge queue.

But _as it is now_, the new case is not added to the `integratedTests` suite.
IIUC you would need some action [here](https://github.com/GEOSX/integratedTests/tree/develop/update/run/compositionalMultiphaseFlow) and [there](https://github.com/GEOSX/integratedTests/blob/develop/update/run/compositionalMultiphaseFlow/compositionalMultiphaseFlow.ats)?
As a matter of fact, this can be merged, but would never get played (I think).

Better wait for @francoishamon for extra information.

**Comment by jhuang2601** [comment]
> > @TotoGaz We need to rebaseline as new smoke test is added into integratedTest. But I cannot run it on Quartz and have requested @francoishamon to do it for me.
> 
> Yes, I was trying to give it a go myself to purge the merge queue.
> 
> But _as it is now_, the new case is not added to the `integratedTests` suite. IIUC you would need some action [here](https://github.com/GEOSX/integratedTests/tree/develop/update/run/compositionalMultiphaseFlow) and [there](https://github.com/GEOSX/integratedTests/blob/develop/update/run/compositionalMultiphaseFlow/compositionalMultiphaseFlow.ats)? As a matter of fact, this can be merged, but would never get played (I think).
> 
> Better wait for @francoishamon for extra information.

Yes, I have created a PR in the integratedTests branch for adding the new smoke test (see [here](https://github.com/GEOSX/integratedTests/pull/248)); but I cannot merge this PR in the integratedTests branch as this new test should be running on Quartz for rebaseline, which needs help from @francoishamon.

**Comment by TotoGaz** [comment]
Oh, OK. I did not see the PR + I could not see the modified `integratedTests` repository modified in the files.

**Comment by TotoGaz** [comment]
@francoishamon I let you deal with it then?

**Comment by francoishamon** [comment]
> @francoishamon I let you deal with it then?

Hello @jhuang2601 and @TotoGaz yes, I am going to work on the rebaseline today

**Pull request merged**